### ISSUE TITLE
Misskey account shows wrong domain on remote users having a conversation on the same remote instance

### ISSUE THREAD

**Issue by trwnh** [tateisu/SubwayTooter#183]
### Description

User `alice@remote.com` is talking with user `bob@remote.com` and I am on `local.com`

ST instead parses the mentions as `@alice@local.com` and `@bob@local.com`

The payload contains `username: alice` and `host: remote.com` but the host does not seem to be used for constructing a mention

### Steps to reproduce

1. View a conversation between `alice@remote.com` and `bob@remote.com`

### Expected behaviour

Mentions show up as `@alice@remote.com` and `@bob@remote.com`, which resolve correctly

### Actual behavior

Mentions show up as `@alice@local.com` and `@bob@local.com`, which do not load because they do not exist

### Misc

- `local.com` is running Misskey 12.100.2
- `remote.com` is currently having a Cloudflare DNS issue

**Comment by trwnh** [comment]
![Screenshot_20220428-035431](https://user-images.githubusercontent.com/10606431/165718106-fccc857a-6b8b-4253-8b69-7f898cf6b7d9.png)

**Comment by tateisu** [comment]
in your example, what domain is misskey? what domain is not ? 

my case problem not reproduced,
- view from misskey.io
- convesations fedibird.com(mastodon base) with mastodon2.juggler.jp(mastodon)
![Screenshot_20220530-215550_Subway Tooter](https://user-images.githubusercontent.com/333944/170997064-e70dc86b-9263-4e79-879f-d6a66d13f880.jpg)

**Comment by trwnh** [comment]
both are misskey, sorry -- i only mentioned the misskey version of nya.social because that was accessible at the time i filed the issue, whereas misskey.bubbletea.dev was down (but still running misskey of course)

**Comment by trwnh** [comment]
to clarify: in the example above, i am on nya.social, and two users on misskey.bubbletea.dev were having a conversation. everyone in the screenshot is from bubbletea.

**Comment by trwnh** [comment]
reproduced just now with https://misskey.bubbletea.dev/notes/90u6bbov45

**Comment by trwnh** [comment]
```
Status-URL: https://misskey.bubbletea.dev/notes/90u6bbov45
Date: 2022-05-28 20:14:28
From-Acct: pie@misskey.bubbletea.dev

@shibao@nya.social @adam@unperson.us wear more clothes! im always dying in summer vecause its impossible to wear any less
Status-Source: {
  "id": "81810d60f9ef76388ae3dd26",
  "createdAt": "2022-05-29T01:14:28.207Z",
  "userId": "817e70a291d87ec3a8261d5b",
  "user": {
    "id": "817e70a291d87ec3a8261d5b",
    "name": "noodle",
    "username": "pie",
    "host": "misskey.bubbletea.dev",
    "avatarUrl": "https://misskey.bubbletea.dev/files/webpublic-480a7db7-32e8-4d03-9a36-c9770f277e1e",
    "avatarBlurhash": "y,N,^;t6~Wofxaj[Iot6ayfRaffRoeWBs=ofbEbFRkRjay%LazRjj[j[ayWUoffPRkjuoLayoft6ayWCoeoffkWBaffia{WCofofWB",
    "avatarColor": null,
    "instance": {
      "name": "bubbletea.dev",
      "softwareName": "misskey",
      "softwareVersion": "12.110.1",
      "iconUrl": "https://misskey.bubbletea.dev/static-assets/icons/192.png",
      "faviconUrl": "https://misskey.bubbletea.dev/files/256d675b-5548-426c-92e4-510c4f09487b",
      "themeColor": "#86b300"
    },
    "emojis": [],
    "onlineStatus": "unknown",
    "_fromStream": false
  },
  "text": "@shibao @adam@unperson.us wear more clothes! im always dying in summer vecause its impossible to wear any less",
  "cw": null,
  "visibility": "public",
  "renoteCount": 0,
  "repliesCount": 1,
  "reactions": {},
  "emojis": [],
  "fileIds": [],
  "files": [],
  "replyId": "81810d60116b9588c5110cb9",
  "renoteId": null,
  "mentions": [
    "817ab2d3c10b822064df6b1c",
    "817fef9ccdd1e944e1571c8e"
  ],
  "uri": "https://misskey.bubbletea.dev/notes/90u6bbov45",
  "reply": {
    "id": "81810d60116b9588c5110cb9",
    "createdAt": "2022-05-29T01:13:28.683Z",
    "userId": "817ab2d3c10b822064df6b1c",
    "user": {
      "id": "817ab2d3c10b822064df6b1c",
      "name": "受不了包",
      "username": "shibao",
      "host": "misskey.bubbletea.dev",
      "avatarUrl": "https://misskey.bubbletea.dev/files/webpublic-f7c7706f-070e-47b8-838f-fdb02b45f7f0",
      "avatarBlurhash": "yFEnt{EN02%0-;f7Io0O-T={EN$zofs.IAI;~BxZ0zn%S5XnWVE2xZ-TR*%1^jxZIVIpozofR*E2NI%2xZRjs,WBe.s-flbIjYI;W=",
      "avatarColor": null,
      "instance": {
        "name": "bubbletea.dev",
        "softwareName": "misskey",
        "softwareVersion": "12.110.1",
        "iconUrl": "https://misskey.bubbletea.dev/static-assets/icons/192.png",
        "faviconUrl": "https://misskey.bubbletea.dev/files/256d675b-5548-426c-92e4-510c4f09487b",
        "themeColor": "#86b300"
      },
      "emojis": [],
      "onlineStatus": "unknown",
      "_fromStream": false
    },
    "text": "@adam@unperson.us i am always too cold in winter 😭",
    "cw": null,
    "visibility": "public",
    "renoteCount": 0,
    "repliesCount": 1,
    "reactions": {},
    "emojis": [],
    "fileIds": [],
    "files": [],
    "replyId": "81810d5d865020bae221c4bc",
    "renoteId": null,
    "mentions": ["817fef9ccdd1e944e1571c8e"],
    "uri": "https://misskey.bubbletea.dev/notes/90u6a1rfft",
    "_fromStream": false
  },
  "_fromStream": false
}
```

what appears to be happening is because of the content not including a domain. subway tooter assumes **your** domain instead of **the author's** domain

**Comment by tateisu** [comment]
fixed at  f57ac32c78c4f32eab7c66d40b668eb62ecc2756

**Comment by tateisu** [comment]
I believe this have been fixed.
